app/assets/javascripts/controllers/ems_common/arbitration_profile_form_controller.js

```diff
@@ -104,30 +97,55 @@ ManageIQ.angular.app.controller('arbitrationProfileFormController', ['$scope', '
+      $scope.cloud_subnets   = response.cloud_subnets;
+      $scope.security_groups = response.security_groups;
+      $scope._cloud_subnet = _.find($scope.cloud_subnets, {id: $scope.arbitrationProfileModel.cloud_subnet_id})
+      $scope._security_group = _.find($scope.security_groups, {id: $scope.arbitrationProfileModel.security_group_id})
     })
 
-    $scope.arbitrationProfileModel.cloud_subnet_id = '';
-    $scope.arbitrationProfileModel.security_group_id = '';
+    if ($scope.arbitrationProfileModel.cloud_network_id != id){
+      $scope.arbitrationProfileModel.cloud_subnet_id = '';
+      $scope.arbitrationProfileModel.security_group_id = '';
+    }
+    $scope.arbitrationProfileModel.cloud_network_id = id;
   };
 
   $scope.profileOptions = function(id, cloud_network_id) {
     var url = "/api/providers/" + id + "?attributes=key_pairs,availability_zones,cloud_networks,cloud_subnets,flavors,security_groups";
 
     API.get(url).then(function(response) {
-      $scope.arbitrationProfileModel.authentications    = response.key_pairs;
-      $scope.arbitrationProfileModel.availability_zones = response.availability_zones;
-      $scope.arbitrationProfileModel.flavors            = response.flavors;
-      $scope.arbitrationProfileModel.cloud_networks     = response.cloud_networks;
+      $scope.authentications    = response.key_pairs;
+      $scope.availability_zones = response.availability_zones;
+      $scope.flavors            = response.flavors;
+      $scope.cloud_networks     = response.cloud_networks;
       if(cloud_network_id != "") {
         $scope.cloudNetworkChanged(cloud_network_id)
       } else {
-        $scope.arbitrationProfileModel.cloud_subnets   = response.cloud_subnets;
-        $scope.arbitrationProfileModel.security_groups = response.security_groups;
+        $scope.cloud_subnets   = response.cloud_subnets;
+        $scope.security_groups = response.security_groups;
       }
+      $scope._authentication = _.find($scope.authentications, {id: $scope.arbitrationProfileModel.authentication_id})
+      $scope._availability_zone = _.find($scope.availability_zones, {id: $scope.arbitrationProfileModel.availability_zone_id})
+      $scope._flavor = _.find($scope.flavors, {id: $scope.arbitrationProfileModel.flavor_id})
+      $scope._cloud_network = _.find($scope.cloud_networks, {id: $scope.arbitrationProfileModel.cloud_network_id})
+      $scope._cloud_subnet = _.find($scope.cloud_subnets, {id: $scope.arbitrationProfileModel.cloud_subnet_id})
+      $scope._security_group = _.find($scope.security_groups, {id: $scope.arbitrationProfileModel.security_group_id})
     })
+
   };
 
+  // watch for all the drop downs on screen
+  "authentication availability_zone flavor cloud_network cloud_subnet security_group".split(" ").forEach(idWatch)
+
+  function idWatch(name) {
+    $scope.$watch('_' + name, function(value) {
+      if (value) {
+        if (name == 'cloud_network')
```

Comment:
I wonder if this would make more sense in a separate watch? (The cloud_network special case) - You can watch a thing more than once..